index.js

```diff
@@ -49,3 +49,21 @@ function compose (middleware) {
     }
   }
 }
+
+/**
+ * composes a wrapped middleware using middleware._wrappers
+ *
+ * @param {Function} middleware
+ * @return {Function} composed
+ * @api public
+ */
+
+function wrap (middleware) {
+  if (!middleware) return false
+  if (!middleware._wrappers) return middleware
+  if (!Array.isArray(middleware._wrappers)) throw new TypeError('._wrappers must be an array')
+  if (!middleware._wrappers.length) return middleware
+  const wrappers = middleware._wrappers
+  delete middleware._wrappers
+  return compose([...wrappers, middleware])
```

Comment:
This isn't how we want the wrappers to work, correct? I think you should do `return wrappers.reduce((mw, wrapper) => wrapper(mw), middeware)`. Also, you should put this in either a separate library or Koa.